packages/uniforms-bootstrap3/src/ListAddField.tsx

```diff
@@ -35,13 +35,22 @@ function ListAdd({
   const limitNotReached =
     !disabled && !(parent.maxCount! <= parent.value!.length);
 
+  const onClickHandler = () => {
+    if (limitNotReached && !readOnly) {
+      parent.onChange(parent.value!.concat([cloneDeep(value)]));
+    }
+  };
+
   return (
     <div
       {...filterDOMProps(props)}
+      role="button"
```

Comment:
@radekmie Please take a look at this file. There are multiple errors were there are `div` elements without semantic meaning with `onClick` handlers which results in multiple different linter errors. There are 2 approaches I could take:
1. Change the `div` to a `button` and fix the styles (I tried it and it's unfortunately not as easy as I would like).
2. Add a proper aria role and keyboard handlers to the element.

This is an example of the 2nd point which I think is elegant enough and we should go with it. There is one thing to keep in mind though: `KeyboardEvent.code` [does not work in IE at all](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code#browser_compatibility). I could probably work around it and add more logic but my first question would be - do we need it?

Also another question as the `onKeyDown` handler would basically be the same in all the files. Do you think it would make sense to export a helper function for it and if yes then what would be an appropriate place for it? Something like this:
```ts
const onKeyDownCreator = (onClickHandler: () => void) => (event: KeyboardEvent) => {
  if (event.code === 'Enter') {
    onClickHandler();
  }
}
```

Reply:
I'd definitely go with 2. As we've already discussed, it's not that easy to get rid of all button-like styles in a cross-browser way.

While this code would be the same for multiple packages, I don't think there's a need to share it - we already do that and not all of the packages need that (see `DateField` for an example and the AntD theme as a counterexample).

I also believe we could have one handler for both cases. something like this:
```tsx
function onAction(event: KeyboardEvent | MouseEvent) {
  if (
    limitNotReached &&
    !readOnly &&
    (!(event instanceof KeyboardEvent) || event.code === 'Enter')
  ) {
    parent.onChange(parent.value!.concat([cloneDeep(value)]));
  }
}
```

When it comes to cross-browser compatibility, I believe we could use [`key`, as it's part of the React `SyntheticEvent` API](https://reactjs.org/docs/events.html#keyboard-events).

Reply:
A few remarks:

The conditional has to look like this:
```ts
(!(event.nativeEvent instanceof KeyboardEvent) || event.key === 'Enter')
```
Which would require me to write a type guard which is another function which might not look nice.

I could also switch this condition to:
```ts
(!('key' in event) || event.key === 'Enter')
```

Which has an additional advantage of not breaking all of our examples in playground/docs. `instanceof KeyboardEvent` returns `false` in all iframe's due to different reference of the prototype chain which would require some more hackery to make this work. 

What do you think? @radekmie

Reply:
Why `event.nativeEvent`? Have you tried importing both `KeyboardEvent` and `MouseEvent` from `react`? It won't cause problems with the iFrame. If not, we could use the one with `'key' in event`.

Reply:
@radekmie Which `KeyboardEvent` and `MouseEvent` do you have in mind? AFAIK there is only a `SyntheticEvent` class which is the same for both mouse and keyboard handlers and the two you mentioned are only typescript types. Or am I mistaken?

Reply:
There's [this](https://github.com/facebook/react/blob/9198a5cec0936a21a5ba194a22fcbac03eba5d1d/packages/react-dom/src/events/SyntheticEvent.js#L472-L474), but I'm not sure. Just try and let me know.

Reply:
I don't think this is even exported from React, couldn't even find it in types and it is `undefined` when accessing `React.SyntheticKeyboardEvent`. Also the event's prototype seems to be just `SyntheticEvent` so I don't think this works.
![image](https://user-images.githubusercontent.com/36732892/111340235-bec96800-8678-11eb-8c1d-43353c1b4de4.png)

Reply:
Changed all checks to `'key' in event` 🦖